### DISCUSSION TITLE
match yarn's property quoting behaviour when writing yarn.lock

### DISCUSSION THREAD

**Discussion by goto-bus-stop** [npm/feedback#309] [CLI] [answered]
On my team some developers prefer using yarn while others prefer npm. My project has used npm 6 thus far, but we were hoping that npm 7 would allow both camps to use their preferred package manager by committing a yarn lockfile, and having npm 7 update it.

It all works beautifully, but we hit a snag due to the different ways yarn and npm write to the yarn.lock file. npm quotes all strings in the output, while yarn only quotes strings that have certain characters in them. The difference means that any time the lockfile is changed by either, (almost) _every_ line in the lockfile changes, only adding or removing quotes. It's a minor annoyance that it ruins the "lines changed" count in PRs, but it also makes dependency changes harder to audit.

Based on the code comments in arborist, it seems intentional:
https://github.com/npm/arborist/blob/9e324976ea5211dc24430a83928ca41b69e46b33/lib/yarn-lock.js#L9-L12

Maybe this decision could be revisited?

**Comment by arcanis** [comment]
> On my team some developers prefer using yarn while others prefer npm

Side note, I want to be clear that this isn't something the Yarn team recommends. There is no way npm and Yarn will be able to guarantee the exact same install layout¹, meaning that some of your developers may face different behaviours than others. Additionally, since npm doesn't support all the Yarn syntaxes or features (and the other way around), it'll lead to further subtle differences of environments when things like `resolutions` will be silently ignored.

I strongly recommend to use either Yarn or npm, but to get everyone to stick to the same one on a given project.

¹ (Which isn't to say that any layout would be incorrect - it's just that the hoisting follows heuristics that are left to their implementers and that will differ from one package manager to another, often in very subtle and hard to debug ways.)

**Reply by goto-bus-stop** [reply]
It does make sense, and is kind of obvious now that you've said it :)

**Reply by goto-bus-stop** [reply]
I guess that does answer my core question (in that we shouldn't go this route). Matching the behaviour could still be nice to reduce the diff for people who do a one-way migration, but I see why that was not deemed worth the effort.

**Comment by MylesBorins** [comment, marked as answer]
hey @goto-bus-stop. This seems like a great thing to open as an issue over in the arborist repo. As you mentioned this behavior seems intentional, but it might be worth the team revisiting for compatibility (the reason to have this feature in the first place)
